### PR TITLE
Tomcat support

### LINKED ISSUE #183: Tomcat app not working on IDEA
### Bug Description

When a user tried running a Tomcat server in IDEA, it started properly but couldn't receive incoming traffic.

### Steps to Reproduce

N/A

### Backtrace

_No response_

### Relevant Logs

_No response_

### Your operating system and version

N/A

### Local process

N/A

### Local process version

_No response_

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by Razz4780**
Closes #183

### CODE REVIEW

File: modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import com.intellij.execution.wsl.target.WslTargetEnvironmentRequest
+import com.intellij.javaee.appServers.run.configuration.RunnerSpecificLocalConfigurationBit
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.components.service
+import com.metalbear.mirrord.CONFIG_ENV_NAME
+import com.metalbear.mirrord.MirrordLogger
+import com.metalbear.mirrord.MirrordProjectService
+import java.util.concurrent.ConcurrentHashMap
+
+private const val DEFAULT_TOMCAT_SERVER_PORT: String = "8005"
```
Comment (meowjesty) [inline modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt:19]: Where did you get this from?

Comment (Razz4780) [reply modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt:19]: From what I've seen online, looks like it's a default shutdown port for Tomcat.

**Review by eyalb181 — APPROVED** [review]

File: modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                mirrordEnv["MIRRORD_IGNORE_DEBUGGER_PORTS"] = getTomcatServerPort()
+
+                savedEnvs[executorId] = envVars.toList()
+                envVars.addAll(mirrordEnv.map { (k, v) -> EnvironmentVariable(k, v, false) })
+                config.setEnvironmentVariables(envVars)
+            } catch (_: Throwable) {
+                // Error notifications were already fired.
+                // We can't abort the execution here, so we let the app run without mirrord.
+                service.notifier.notifySimple(
+                    "Cannot abort run due to platform limitations, running without mirrord",
```
Comment (eyalb181) [inline modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt:70]: Does the user know there is an error at this point, i.e. why we are trying to abort the run in the first place?

Comment (Razz4780) [reply modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt:70]: Yup, at this point notifications were shown

Comment (eyalb181) [reply modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt:70]: Ah sorry, you wrote that

**Pull request merged**